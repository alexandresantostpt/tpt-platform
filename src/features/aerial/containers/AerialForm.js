import { connect } from 'react-redux'

import { getObjSelector as travelGetObjSelector, getClientsSelector } from '@features/travelScript/selectors'

import { requestSearchCityOurApiAction } from '@features/travel/actions'
import { getSuggestionsSelector } from '@features/travel/selectors'

import { getObjSelector } from '../selectors'
import { requestDelAction, requestGetSingleAction, requestSaveDataAction } from '../actions'

import AerialForm from '../pages/AerialForm'

const mapStateToProps = state => ({
    aerial: getObjSelector(state),
    clients: getClientsSelector(state),
    suggestions: getSuggestionsSelector(state),
    travelScript: travelGetObjSelector(state)
})

const mapDispatchToProps = {
    requestDelAction,
    requestGetSingleAction,
    requestSaveDataAction,
    requestSearchCityOurApiAction
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AerialForm)
