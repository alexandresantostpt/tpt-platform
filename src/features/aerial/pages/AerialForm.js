import { events } from '@constants/events'

import PropTypes from 'prop-types'
import PubSub from 'pubsub-js'
import QueryString from 'query-string'
import React, { useEffect, useRef, useState } from 'react'

import { colors } from '@helpers/colors'
import { i18n } from '@i18n'
import { services } from '@/constants/services'
import { withRouter } from 'react-router-dom'
import { replaceEnterWithSpace } from '@utils/string'

import Button from '@components/button/Button'
import Cancel from '@components/cancel/Cancel'
import FormBuilder from '@components/form/FormBuilder'
import Panel from '@components/panel/Panel'

import FormButtonContainer from '@components/form/container/button/FormButtonContainer'

import AerialServiceModel from '../models/AerialServiceModel'
import AerialClientForm from '../components/aerialClientForm/AerialClientForm'
import Tabs from '../components/tabs/Tabs'

import { aerialFormFields } from '../fields'

const AerialList = props => {
    const { aerial, match, requestSearchCityOurApiAction, suggestions, travelScript } = props
    const {
        params: { id, idAerial }
    } = match
    const { dateId, scriptDate, scriptId } = QueryString.parse(location.search)

    const previousPath = `/travel/${id}/script`
    const form = useRef()

    const [activeTab, updateActiveTab] = useState(0)
    const [serviceDisabled, updateServiceDisabled] = useState(false)
    const [image, updateImage] = useState()
    const [parts, updateParts] = useState([aerialFormFields(0)])

    useEffect(() => {
        const { requestGetSingleAction } = props
        if (idAerial) {
            requestGetSingleAction({ id: idAerial })
            updateServiceDisabled(true)
        } else {
            requestGetSingleAction({ id: null })
            updateServiceDisabled(false)
        }
    }, [idAerial])

    useEffect(() => {
        if (aerial && aerial.id === idAerial) {
            updateParts([...aerial.toForm().journeys.map((_, index) => aerialFormFields(index))])
        } else {
            updateParts([aerialFormFields(0)])
        }
    }, [aerial])

    const removePartsLastItem = () => {
        const newParts = [...parts]
        newParts.pop()
        updateParts(newParts)
    }

    const callSaveData = values => {
        values.journeys.map(journey => (journey.flightNumber = replaceEnterWithSpace(journey.flightNumber)))
        const { requestSaveDataAction } = props
        requestSaveDataAction({
            data: {
                dateId,
                scriptId,
                service: new AerialServiceModel({ ...values, scriptDate }, true).toDTO(),
                type: services.AERIAL
            },
            image,
            redirectTo: previousPath
        })
        reset()
    }

    const onSaveClick = () => form.current.form.submit()

    const reset = () => {
        PubSub.publish(events.FORM_RESET)
        updateParts([aerialFormFields(0)])
        updateImage(null)
    }
    return (
        <>
            <Panel
                backgroundImageUrl={aerial && aerial.getImage()}
                icon="plane-alt"
                id={aerial && aerial.id}
                uploadImage={updateImage}
            >
                <Tabs
                    activeTab={activeTab}
                    parts={parts}
                    removePartsLastItem={removePartsLastItem}
                    updateActiveTab={updateActiveTab}
                    updateParts={updateParts}
                />
            </Panel>
            <FormBuilder
                clients={props.clients}
                disabled={serviceDisabled}
                fieldChildren={<AerialClientForm clients={props.clients} disabled={serviceDisabled} form={form} />}
                fields={parts[activeTab]}
                fieldsToValidate={parts}
                getSuggestions={requestSearchCityOurApiAction}
                initialValues={(aerial && aerial.toForm()) || { values: [{ dateExit: scriptDate }] }}
                onSubmit={callSaveData}
                rangeDates={travelScript && travelScript.getRangeDates()}
                ref={process.env.__TEST__ ? null : form}
                suggestions={suggestions}
            />
            <FormButtonContainer>
                <Button
                    color={colors.purple}
                    onClick={() => (serviceDisabled ? updateServiceDisabled(false) : onSaveClick())}
                    text={i18n.t(serviceDisabled ? 'buttons.editService' : 'buttons.saveService')}
                />
                <Cancel link={previousPath} />
            </FormButtonContainer>
        </>
    )
}

AerialList.defaultProps = {
    aerial: {},
    clients: []
}

AerialList.propTypes = {
    aerial: PropTypes.objectOf(AerialServiceModel),
    clients: PropTypes.array,
    requestDelAction: PropTypes.func.isRequired,
    requestGetSingleAction: PropTypes.func.isRequired,
    requestSaveDataAction: PropTypes.func.isRequired,
    requestSearchCityOurApiAction: PropTypes.func.isRequired,
    suggestions: PropTypes.array
}

export default withRouter(AerialList)
