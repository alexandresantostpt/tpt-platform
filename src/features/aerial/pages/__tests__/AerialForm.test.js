import React from 'react'
import renderer from 'react-test-renderer'

import AerialForm from '../AerialForm'

import AerialServiceModel from '../../models/AerialServiceModel'
import TravelScriptModel from '@/features/travelScript/models/TravelScriptModel'

describe('Tests for Travel component', () => {
    const aerial = new AerialServiceModel({
        __v: 2,
        _id: '5c9a76ee9be6c0345f00c653',
        createdAt: '2019-03-26T19:01:02.109Z',
        journeys: [
            {
                _id: '5c9a79759be6c0231400c658',
                companyName: 'Companhia de testes',
                flightNumber: '123',
                from: { airport: 'teste', city: 'teste', date: '2019-03-27T15:00:00.000Z', hour: '2019-03-26T19:00:00.696Z' },
                passengers: [
                    {
                        _id: '5c9a76ee9be6c00ed700c655',
                        client: '5c9a17959be6c0225700c5ed',
                        reserveCode: '12313',
                        seat: 'teste2',
                        session: 'luxo',
                        ticket: 12313
                    },
                    {
                        _id: '5c9a79759be6c09ba500c659',
                        client: '5c9a24f29be6c0bd9f00c627',
                        reserveCode: 't123s',
                        seat: 'Assento 22b',
                        session: 'teste',
                        ticket: 123123
                    }
                ],
                to: { airport: 'teste', city: 'teste', hour: '2019-03-26T21:30:00.464Z' }
            }
        ],
        script: '5c9a17959be6c0c46200c5ef',
        scriptDate: '2019-03-26T00:00:00.000Z',
        type: 'AERIAL',
        updatedAt: '2019-03-26T19:11:49.707Z'
    })
    const clients = [
        {
            __v: 0,
            _id: '5c9a17959be6c0225700c5ed',
            active: false,
            blocked: false,
            cpf: '123.123.123-12',
            createdAt: '2019-03-26T12:14:13.525Z',
            email: 'teste@teste.com',
            name: 'Viajante 1',
            password: '1b621fe4e8bd896fa918088696a393ab2330ee73373efe770833b85598756a28',
            updatedAt: '2019-03-26T13:15:28.197Z'
        },
        {
            __v: 0,
            _id: '5c9a24f29be6c0bd9f00c627',
            active: false,
            blocked: false,
            cpf: '456.456.986-22',
            createdAt: '2019-03-26T13:11:14.605Z',
            email: 'ts2@ts.com',
            name: 'Ceccon TEST2',
            password: '5f8fd7b0e272f3061fe4e4a399e529e4f5076f9f55389bd8a40fe32d9521e967',
            updatedAt: '2019-03-26T13:46:12.914Z'
        }
    ]
    const travelScript = new TravelScriptModel({ travel: { __v: 0 } })

    it('Should render withou crash', () => {
        const match = {
            params: {
                id: '5c52ebcd5fc7c949223ab705'
            }
        }
        const tree = renderer
            .create(
                <AerialForm
                    aerial={aerial}
                    clients={clients}
                    match={match}
                    requestDelAction={jest.fn()}
                    requestGetSingleAction={jest.fn()}
                    travelScript={travelScript}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
