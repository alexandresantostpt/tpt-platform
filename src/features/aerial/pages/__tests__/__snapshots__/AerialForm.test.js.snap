// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Travel component Should render withou crash 1`] = `
Array [
  <div
    className="Panel__StyledPanel-sc-1qzkpb6-0 cGUlCn"
  >
    <span
      className="Panel__StyledIcon-sc-1qzkpb6-2 iavvSk"
    >
      <span
        className="icon icon-plane-alt Icon__StyledIcon-crr1cg-0 dfhcZR"
        disabled={false}
        fontSize="3rem"
        onClick={null}
      />
    </span>
    <div
      className="UploadButton__StyledContainer-sc-1p8zoc-0 fpZFIG"
    >
      <label
        className="UploadButton__StyledUploadButton-sc-1p8zoc-3 ivhrYW"
        htmlFor="inputFile"
      >
        +
      </label>
      <input
        accept="image/png, image/jpeg"
        className="UploadButton__InputFile-sc-1p8zoc-4 eLaooA"
        id="inputFile"
        onChange={[Function]}
        type="file"
      />
    </div>
  </div>,
  <ul
    className="Tabs__StyledTabs-sc-17hg029-0 fiSwC"
  >
    <li
      className="Tab__StyledTab-sc-1hgxzl3-0 cykjJO"
      onClick={null}
    >
      <span
        className="TabTitle__StyledTabTitle-sc-1ywdo8e-0 dKrIjz"
        onClick={[Function]}
      >
        Trecho  1
      </span>
      <span
        className="icon icon-times Icon__StyledIcon-crr1cg-0 dqcjNO"
        disabled={false}
        fontSize="1.5rem"
        onClick={[Function]}
      />
    </li>
    <li
      className="Tab__StyledTab-sc-1hgxzl3-0 feGFRG"
      onClick={[Function]}
    >
      <span
        className="TabTitle__StyledTabTitle-sc-1ywdo8e-0 dKrIjz"
        onClick={null}
      >
        Adicionar trecho
      </span>
    </li>
  </ul>,
  <form
    className="FormBuilder__StyledForm-c1htou-0 PZxQa"
    onSubmit={[Function]}
  >
    <div
      className="form__container"
    >
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Nome da companhia
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-journeys[0].companyName"
          disabled={false}
          id="journeys[0].companyName"
          maxLength={120}
          name="journeys[0].companyName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Companhia de testes"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Número do voo
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <textarea
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormTextArea__StyledFormText-sc-9g3u2c-0 hmDUjT"
          data-cy="input-journeys[0].flightNumber"
          disabled={false}
          id="journeys[0].flightNumber"
          isInvalid={false}
          min={0}
          name="journeys[0].flightNumber"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="JJ
Número do voo"
          readOnly={false}
          required={true}
          type="text"
          value="123"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Cidade de saída
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormCity__StyledAutosuggest-sc-4vmpmt-0 kHEvhJ"
        >
          <div
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-1"
            className="react-autosuggest__container"
            role="combobox"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls="react-autowhatever-1"
              autoComplete="off"
              className="react-autosuggest__input"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
            <div
              className="react-autosuggest__suggestions-container"
              id="react-autowhatever-1"
              role="listbox"
            />
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Aeroporto de saída
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-journeys[0].airportExit"
          disabled={false}
          id="journeys[0].airportExit"
          maxLength={60}
          name="journeys[0].airportExit"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="teste"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Data de saída
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormDate__StyledFormDate-sc-1cgz0rk-0 ebKpCI"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-"
                    aria-label="Selecione uma data..."
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                    disabled={false}
                    id=""
                    name=""
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Selecione uma data..."
                    readOnly={true}
                    required={false}
                    type="text"
                    value="27 Mar 2019"
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon icon-calendar Icon__StyledIcon-crr1cg-0 cmjVJQ"
                    disabled={false}
                    fontSize="1.75rem"
                    onClick={null}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Horário de saída
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="16:00"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Cidade de destino
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormCity__StyledAutosuggest-sc-4vmpmt-0 kHEvhJ"
        >
          <div
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-1"
            className="react-autosuggest__container"
            role="combobox"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls="react-autowhatever-1"
              autoComplete="off"
              className="react-autosuggest__input"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
            <div
              className="react-autosuggest__suggestions-container"
              id="react-autowhatever-1"
              role="listbox"
            />
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Aeroporto de destino
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-journeys[0].airportDestiny"
          disabled={false}
          id="journeys[0].airportDestiny"
          maxLength={60}
          name="journeys[0].airportDestiny"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="teste"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Data de chegada
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormDate__StyledFormDate-sc-1cgz0rk-0 ebKpCI"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-"
                    aria-label="Selecione uma data..."
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                    disabled={false}
                    id=""
                    name=""
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Selecione uma data..."
                    readOnly={true}
                    required={false}
                    type="text"
                    value=""
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon icon-calendar Icon__StyledIcon-crr1cg-0 cmjVJQ"
                    disabled={false}
                    fontSize="1.75rem"
                    onClick={null}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Horário de chegada
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="18:30"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Passageiro
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormClients__StyledOption-jb3pw2-1 gugIhy"
        >
          <input
            checked={true}
            disabled={false}
            id="5c9a17959be6c0225700c5ed"
            name="5c9a17959be6c0225700c5ed"
            onChange={[Function]}
            type="checkbox"
            value="5c9a17959be6c0225700c5ed"
          />
          <span
            className="FormClients__StyledLabel-jb3pw2-0 kKzFxc"
          >
            Viajante 1
          </span>
        </div>
        <div
          className="FormClients__StyledOption-jb3pw2-1 gugIhy"
        >
          <input
            checked={true}
            disabled={false}
            id="5c9a24f29be6c0bd9f00c627"
            name="5c9a24f29be6c0bd9f00c627"
            onChange={[Function]}
            type="checkbox"
            value="5c9a24f29be6c0bd9f00c627"
          />
          <span
            className="FormClients__StyledLabel-jb3pw2-0 kKzFxc"
          >
            Ceccon TEST2
          </span>
        </div>
        <div
          className="AerialClientForm__StyledAerialClientForm-sc-14liczv-0 dMvCuG"
        >
          <span>
            <div
              className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
            >
              <label
                className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
              >
                Nome do passageiro
                 1
              </label>
              <input
                className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                data-cy="input-journeys[0].passengers[0].passengerName"
                disabled={false}
                id="journeys[0].passengers[0].passengerName"
                maxLength={60}
                name="journeys[0].passengers[0].passengerName"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder=""
                readOnly={false}
                required={false}
                type="text"
                value="Viajante 1"
              />
            </div>
            <div
              className="Collapse__StyledCollapseItem-sc-16jlua7-0 hveKjh"
            >
              <div
                className="CollapseItems__StyledCollapseItems-tyowld-0 fnYIc"
              >
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    Assento
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-journeys[0].passengers[0].seat"
                    disabled={false}
                    id="journeys[0].passengers[0].seat"
                    maxLength={15}
                    name="journeys[0].passengers[0].seat"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="text"
                    value="teste2"
                  />
                </div>
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    Código da reserva
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-journeys[0].passengers[0].reserveCode"
                    disabled={false}
                    id="journeys[0].passengers[0].reserveCode"
                    maxLength={15}
                    name="journeys[0].passengers[0].reserveCode"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="text"
                    value="12313"
                  />
                </div>
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    E-ticket
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-journeys[0].passengers[0].ticket"
                    disabled={false}
                    id="journeys[0].passengers[0].ticket"
                    maxLength={255}
                    min={0}
                    name="journeys[0].passengers[0].ticket"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="number"
                    value={12313}
                  />
                </div>
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    Classe
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-journeys[0].passengers[0].session"
                    disabled={false}
                    id="journeys[0].passengers[0].session"
                    maxLength={15}
                    name="journeys[0].passengers[0].session"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="text"
                    value="luxo"
                  />
                </div>
              </div>
              <span
                className="icon icon-arrow-up Icon__StyledIcon-crr1cg-0 cMlgiC"
                disabled={false}
                fontSize="1.75rem"
                onClick={[Function]}
              />
            </div>
          </span>
          <span>
            <div
              className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
            >
              <label
                className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
              >
                Nome do passageiro
                 2
              </label>
              <input
                className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                data-cy="input-journeys[0].passengers[1].passengerName"
                disabled={false}
                id="journeys[0].passengers[1].passengerName"
                maxLength={60}
                name="journeys[0].passengers[1].passengerName"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder=""
                readOnly={false}
                required={false}
                type="text"
                value="Ceccon TEST2"
              />
            </div>
            <div
              className="Collapse__StyledCollapseItem-sc-16jlua7-0 hveKjh"
            >
              <div
                className="CollapseItems__StyledCollapseItems-tyowld-0 fnYIc"
              >
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    Assento
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-journeys[0].passengers[1].seat"
                    disabled={false}
                    id="journeys[0].passengers[1].seat"
                    maxLength={15}
                    name="journeys[0].passengers[1].seat"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="text"
                    value="Assento 22b"
                  />
                </div>
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    Código da reserva
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-journeys[0].passengers[1].reserveCode"
                    disabled={false}
                    id="journeys[0].passengers[1].reserveCode"
                    maxLength={15}
                    name="journeys[0].passengers[1].reserveCode"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="text"
                    value="t123s"
                  />
                </div>
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    E-ticket
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-journeys[0].passengers[1].ticket"
                    disabled={false}
                    id="journeys[0].passengers[1].ticket"
                    maxLength={255}
                    min={0}
                    name="journeys[0].passengers[1].ticket"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="number"
                    value={123123}
                  />
                </div>
                <div
                  className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
                >
                  <label
                    className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
                  >
                    Classe
                    
                  </label>
                  <input
                    className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
                    data-cy="input-journeys[0].passengers[1].session"
                    disabled={false}
                    id="journeys[0].passengers[1].session"
                    maxLength={15}
                    name="journeys[0].passengers[1].session"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    readOnly={false}
                    required={false}
                    type="text"
                    value="teste"
                  />
                </div>
              </div>
              <span
                className="icon icon-arrow-up Icon__StyledIcon-crr1cg-0 cMlgiC"
                disabled={false}
                fontSize="1.75rem"
                onClick={[Function]}
              />
            </div>
          </span>
        </div>
      </div>
    </div>
  </form>,
  <div
    className="FormButtonContainer-sc-1cqqivn-0 ADyLt"
  >
    <button
      className="Button__StyledButton-sc-1fc3klx-0 bEGBgE"
      disabled={false}
      fill={false}
      onClick={[Function]}
    >
      Salvar serviço
    </button>
    <a
      className="Cancel__StyledCancel-sc-1prgtit-0 WzuoR"
      to="/travel/5c52ebcd5fc7c949223ab705/script"
    >
      Cancelar
    </a>
  </div>,
]
`;
